samplers/aws/xray/internal/manifest.go

```diff
@@ -0,0 +1,352 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "go.opentelemetry.io/contrib/samplers/aws/xray/internal"
+
+import (
+	"context"
+	crypto "crypto/rand"
+	"fmt"
+	"math"
+	"net/url"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jinzhu/copier"
+
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+const manifestTTL = 3600
+
+// Manifest represents a full sampling ruleset and provides
+// option for configuring Logger, Clock and xrayClient.
+type Manifest struct {
+	Rules                          []Rule
+	SamplingTargetsPollingInterval time.Duration
+	refreshedAt                    time.Time
+	xrayClient                     *xrayClient
```

Comment:
Nit:  This should probably be an interface, so you don't have to create an http server etc to test your manifest code.  It could be an internal interface that the xray client Already meets.

Reply:
not sure if I follow you on this. I think we need http server to mimic `getSamplingRules` and `getSamplingTargets` API response.